src/machine.rs

```diff
@@ -348,21 +413,76 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'tcx> {
     #[inline(always)]
     fn stack_push(
         ecx: &mut InterpCx<'mir, 'tcx, Self>,
-    ) -> InterpResult<'tcx, stacked_borrows::CallId> {
-        Ok(ecx.memory().extra.stacked_borrows.borrow_mut().new_call())
+    ) -> InterpResult<'tcx, FrameData<'tcx>> {
+        Ok(FrameData {
+            call_id: ecx.memory().extra.stacked_borrows.borrow_mut().new_call(),
+            catch_panic: None,
+            is_box_me_frame: false
+        })
     }
 
     #[inline(always)]
     fn stack_pop(
         ecx: &mut InterpCx<'mir, 'tcx, Self>,
-        extra: stacked_borrows::CallId,
-    ) -> InterpResult<'tcx> {
-        Ok(ecx
-            .memory()
-            .extra
-            .stacked_borrows
-            .borrow_mut()
-            .end_call(extra))
+        extra: FrameData<'tcx>,
+    ) -> InterpResult<'tcx, StackPopInfo> {
+        let prev_unwinding = ecx.machine.unwinding;
+        if extra.is_box_me_frame {
+            trace!("unwinding: found box_me_frame");
+            ecx.machine.unwinding = true;
```

Comment:
I assume in this branch we know we'll return `StartUnwinding`? We could just compute that here instead of indirecting via the state change of the boolean, no?